### PR TITLE
Adds a simple LogMelFrontend.

### PR DESCRIPTION
**Pull request by markblee**
Simple log-Mel frontend, with reasonable default values for most of the stages. If needed, we can make certain stages configurable in a future PR.

### CODE REVIEW

File: axlearn/audio/frontend.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        # Number of filters/bands in the output spectrogram.
+        num_filters: Required[int] = REQUIRED
+        # Number of input samples per second, e.g., 24000 for 24KHz inputs.
+        sample_rate: Required[int] = REQUIRED
+        # Size of each frame in ms.
+        frame_size_ms: Required[float] = REQUIRED
+        # Hop size in ms.
+        hop_size_ms: Required[float] = REQUIRED
+        # Optional output scaling.
+        scaling: Optional[InstantiableConfig[Callable[[Tensor], Tensor]]] = None
```
Comment (ruomingp) [inline axlearn/audio/frontend.py:67]:
```suggestion
        output_transformation: Optional[InstantiableConfig[Callable[[Tensor], Tensor]]] = None
```

File: axlearn/audio/frontend.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+        # Number of filters/bands in the output spectrogram.
+        num_filters: Required[int] = REQUIRED
+        # Number of input samples per second, e.g., 24000 for 24KHz inputs.
+        sample_rate: Required[int] = REQUIRED
+        # Size of each frame in ms.
+        frame_size_ms: Required[float] = REQUIRED
+        # Hop size in ms.
+        hop_size_ms: Required[float] = REQUIRED
+        # Optional output scaling.
```
Comment (ruomingp) [inline axlearn/audio/frontend.py:66]: Give an example?

File: axlearn/audio/frontend.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    InstantiableConfig,
+    Required,
+    config_class,
+    maybe_instantiate,
+)
+from axlearn.common.module import Module
+from axlearn.common.utils import Tensor
+
+
+def scale_by_mean_std(
```
Comment (ruomingp) [inline axlearn/audio/frontend.py:31]: Looks like this is not just scaling, but also shifting. Is `normalize_by_mean_std` a more accurate name?

File: axlearn/audio/frontend.py (hunk 1 above)
Comment (markblee) [reply axlearn/audio/frontend.py]: Done.

File: axlearn/audio/frontend.py (hunk 2 above)
Comment (markblee) [reply axlearn/audio/frontend.py]: Done.

File: axlearn/audio/frontend.py (hunk 3 above)
Comment (markblee) [reply axlearn/audio/frontend.py]: Sounds good, updated.

**Review by ruomingp — APPROVED** [review]

**Pull request merged**